cmds/tcz/tcz.go

```diff
@@ -127,42 +95,32 @@ func fetch(p string) error {
 	packageName := filepath.Join(tczServerDir, p)
 
 	if _, err := os.Stat(fullpath); !os.IsNotExist(err) {
-		debug("package %s is downloaded\n", fullpath)
-		return nil
+		// Either already exists (already been downloaded) or some
+		// unresolvable error.
+		return err
 	}
 
-	if _, err := os.Stat(fullpath); err != nil {
-		cmd := fmt.Sprintf("http://%s:%s/%s", *host, *port, packageName)
-		debug("Fetch %v\n", cmd)
-
-		resp, err := http.Get(cmd)
-		if err != nil {
-			l.Fatalf("Get of %v failed: %v\n", cmd, err)
-		}
-		defer resp.Body.Close()
+	cmd := fmt.Sprintf("http://%s:%s/%s", *host, *port, packageName)
+	resp, err := http.Get(cmd)
+	if err != nil {
+		return fmt.Errorf("http.Get(%s) failed: %v", cmd, err)
+	}
+	defer resp.Body.Close()
 
-		if resp.Status != "200 OK" {
-			debug("%v Not OK! %v\n", cmd, resp.Status)
-			return syscall.ENOENT
-		}
+	if resp.Status != "200 OK" {
+		return syscall.ENOENT
```

Comment:
I'd argue we should use fmt.Errorf here as well, or at least os.ErrNotExist and not syscall.